app/models/user.rb

```diff
@@ -11,6 +11,18 @@ class User < ActiveRecord::Base
   has_many :locations, through: :entries
   has_many :photos, through: :entries
 
+  has_many :friendships
+  has_many :passive_friendships, class_name: "Friendship", foreign_key: "friend_id"
+
+  has_many :active_friends, -> { where(friendships: { accepted: true}) }, :through => :friendships, :source => :friend
+  has_many :passive_friends, -> { where(friendships: { accepted: true}) }, :through => :passive_friendships, :source => :user
+  has_many :pending_friends, -> { where(friendships: { accepted: false}) }, :through => :friendships, :source => :friend
+  has_many :requested_friendships, -> { where(friendships: { accepted: false}) }, :through => :passive_friendships, :source => :user
```

Comment:
Line is too long. [132/80]<br>Use the new Ruby 1.9 hash syntax.<br>Space inside } missing.